http-client/src/main/java/io/airlift/http/client/HttpUriBuilder.java

```diff
@@ -68,6 +68,8 @@ public static HttpUriBuilder uriBuilder()
     public static HttpUriBuilder uriBuilderFrom(URI uri)
     {
         requireNonNull(uri, "uri is null");
+        checkArgument(uri.getScheme() != null, "URI does not have a scheme: %s", uri);
+        checkArgument(uri.getHost() != null, "URI does not have a host: %s", uri);
```

Comment:
Is there much advantage of verifying early, not in `build()`?

usually, we don't validate in builders' setters, deferring until final call.
This works since builders are typically short-lived.

Reply:
The reason to verify here is that we can print the invalid URI object. We couldn't think of a valid use case to create an `HttpUriBuilder` from a URI where the scheme or host is not set (as that would be something like `http://` or a completely different type of URI ... the JDK bug not withstanding).